src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker/Product.java

```diff
@@ -0,0 +1,206 @@
+     * @param state the state to set it to.
+     */
+    private void setState(final ProductState state)
+    {
+        this.state = state;
+    }
+
+    /**
+     * Getter for the end product of the bakery.
+     * @return the itemStack of the product.
+     */
+    public ItemStack getEndProduct()
+    {
+        return endProduct;
+    }
+
+    /**
+     * Get the Baking progress of the product in the oven.
+     * @return an integer between 0-10.
+     */
+    public int getBakingProgress()
+    {
+        return bakingProgress;
+    }
+
+    /**
+     * Increase the baking progress.
+     * If reached finish line, this will make the product reach the next state.
+     */
+    public void increaseBakingProgress()
+    {
+        this.bakingProgress++;
+
+        if(bakingProgress >= FINISHED_BAKING_PROGRESS)
+        {
+            this.nextState();
+        }
+    }
+
+    /**
+     * Getter for the recipe id of the product.
+     * @return the id.
+     */
+    public int getRecipeId()
+    {
+        return recipeId;
+    }
+
+    /**
+     * Create the product from NBT.
+     * @param productCompound the compound to use.
+     * @return the restored Product.
+     */
+    public static Product createFromNBT(final NBTTagCompound productCompound)
+    {
+        if(productCompound.hasKey(TAG_STATE))
+        {
+            final ProductState state = ProductState.values()[productCompound.getInteger(TAG_STATE)];
+            final int recipeId = productCompound.getInteger(TAG_RECIPE_ID);
+            final Product product = new Product(ItemStack.loadItemStackFromNBT(productCompound), recipeId);
+            product.setState(state);
+            return product;
+        }
+        return null;
+    }
+
+    /**
+     * Write the Product to NBT.
+     * @param productCompound the compound to write it to.
+     */
+    public void writeToNBT(final NBTTagCompound productCompound)
+    {
+        productCompound.setInteger(TAG_STATE, state.ordinal());
+        productCompound.setInteger(TAG_RECIPE_ID, recipeId);
+        endProduct.writeToNBT(productCompound);
+    }
+
+    @Override
+    public boolean equals(final Object o)
+    {
+        if (this == o)
+        {
+            return true;
+        }
+        if (!(o instanceof Product))
+        {
+            return false;
+        }
+
+        final Product product = (Product) o;
+
+        if (recipeId != product.recipeId)
+        {
+            return false;
+        }
+        if (state != product.state)
+        {
+            return false;
+        }
+        if (!endProduct.equals(product.endProduct))
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        int result = state.hashCode();
+        result = 31 * result + endProduct.writeToNBT(new NBTTagCompound()).hashCode();
+        result = 31 * result + recipeId;
+        return result;
+    }
+
+    public enum ProductState
```

Comment:
Enum in a different file.
If you change this to a generic production result class, this needs to be changed. An enum cannot be Extended by external parties, so make it a marker Interface, implement in on this enum. That way we are not forced to use the enum in the future.